docs/guide/checking_nan.rst

```diff
@@ -0,0 +1,188 @@
+Dealing with NaNs and infs
+==========================
+
+During the training of a model on a given environment, it is possible that the RL model becomes completely
+corrupted when a NaN or an inf is given or returned from the RL model.
+
+How and why?
+------------
+
+The issue arises then NaNs or infs do not crash, but simply get propagated through the training,
+until all the floating point number converge to NaN or inf. This is in line with the
+`IEEE Standard for Floating-Point Arithmetic (IEEE 754) <https://ieeexplore.ieee.org/document/4610935>`_ standard, as it says:
+
+.. note::
+    Five possible exceptions can occur:
+        - Invalid operation (:math:`\sqrt{-1}`, :math:`\inf \times 1`, :math:`\text{NaN}\ \mathrm{mod}\ 1`, ...) return NaN
+        - Division by zero:
+            - if the operand is not zero (:math:`1/0`, :math:`-2/0`, ...) returns :math:`\pm\inf`
+            - if the operand is zero (:math:`0/0`) returns signaling NaN
+        - Overflow (exponent too high to represent) returns :math:`\pm\inf`
+        - Underflow (exponent too low to represent) returns :math:`0`
+        - Inexact (not representable exactly in base 2, eg: :math:`1/5`) returns the rounded value (ex: :code:`assert (1/5) * 3 == 0.6000000000000001`)
+
+And of these, only ``Division by zero`` will signal an exception, the rest will propagate invalid values quietly.
+
+In python, dividing by zero will indeed raise the exception: ``ZeroDivisionError: float division by zero``,
+but ignores the rest.
+
+The default in numpy, will warn: ``RuntimeWarning: invalid value encountered``
+but will not halt the code.
+
+And the worst of all, Tensorflow will not signal anything
```

Comment:
dammit ^^

Reply:
yeah, this one shocked me, div zero not being a problem at all